### DISCUSSION TITLE
Inputs provided by extension?

### DISCUSSION THREAD

**Discussion by ClockworkV** [microsoft/vscode-discussions#321] [Extension Development QnA]
Is there a way to define inputs for tasks provided by my extension?
I can do it in a tasks.json file, and my extension generated tasks can use inputs if they're already in the tasks.json file, but I haven't found a way of generating inputs on the fly like I do with the tasks.
The two ways I have found to do this are a custom execution (which I haven't tried yet) or using a command input with the command being provided by my extension. In the command I can pop up a QuickPick, or any other relevant mechanism of getting my inputs, and return that string. This works just fine, and is even more flexible and powerful than Inputs, but it _is_ extra work, and I feel that for completeness and symmetry we should be able to define Inputs as part of the extension.

**Comment by burkeholland** [comment]
Closing as inactive. If you found a solution for this, please feel free to reopen with the answer and mark as such.
